### PR TITLE
Feature/schedule page

### LINKED ISSUE #3: Feature/Schedule page
- [x] Create pages/Schedule folder
- [x] Create index.tsx for layouting, and content.tsx for content
- [x] Focus on content page, because the layout will be implement after #1

### PR DESCRIPTION
**Pull request by Rendy752**
- **Prisma Online**
image

- **Proof of Filling**
1. Table
image
2. List
image

- **Schedule**
1. Regular
image

2. Daily
image

3. Weekly
image

### CODE REVIEW

**Comment by Rendy752** [comment]
- Add "**react-hot-toast**" package for join or leave course in prisma online
- I think there're a problem in table styling or content wrapper that cause the template content overflow in horizontal direction
image

**Comment by oohgan** [comment]
lgtm

File: src/api/course/model.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  name: string;
+  sks: number;
+  class: string;
+  capacity: number;
+  schedule: ScheduleModel[];
+};
+
+export type SelectedCourseModel = {
+  course: CourseModel;
+  lecturer_note?: string;
```
Comment (oohgan) [inline src/api/course/model.ts:16]: ini knp nullable?, untuk string, harusny empty string kalo katek note

File: src/api/course/service.ts [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+  SelectedCourseModel,
+} from "./model";
+import studentData from "../student/data";
+import { dayName, sessionName } from "@utils/consts";
+import toast from "react-hot-toast";
+import { ScheduleModel } from "../schedule/model";
+
+const selectedCourseData: SelectedCourseLocalStorageModel[] = JSON.parse(
+  localStorage.getItem("selectedCourseData") || "[]",
+);
```
Comment (oohgan) [inline src/api/course/service.ts:14]: bisa dipindah ke custom reusable function untuk handle kalo fail parse

File: src/common/utils/consts.ts [hunk 3]
```diff
@@ -1,6 +1,34 @@
...
+  REJECTED: "Ditolak",
+  NOT_TAKEN: "Tidak diambil",
+};
+
+const statusDetail: { [key: number]: string | string[] } = {
+  1: [statusName.WAITING, "bg-yellow-300"],
+  2: [statusName.APPROVED, "bg-blue-300"],
+  3: [statusName.REJECTED, "bg-red-300"],
+  4: [statusName.NOT_TAKEN, "bg-gray-300"],
+};
```
Comment (oohgan) [inline src/common/utils/consts.ts:32]: ini harusny dipisah consts untuk angka -> string status name, trus status name -> color

File: src/pages/PrismaOnline/partials/CourseList.tsx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    "no",
+    "mata kuliah",
+    "kelas",
+    "jumlah sks",
+    "semester",
+    "kapasitas kelas",
+    "jadwal kuliah",
+    "status",
+    "aksi",
+  ];
```
Comment (oohgan) [inline src/pages/PrismaOnline/partials/CourseList.tsx:53]: logic pindahin ke custom hooks

File: src/pages/PrismaOnline/partials/CourseList.tsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+                  </td>
+                  <td className="px-4 border border-gray-200">
+                    {row.sks ? row.sks : "-"} sks
+                  </td>
+                  <td className="px-4 border border-gray-200">
+                    {row.semester_id ? row.semester_id : "-"}
+                  </td>
+                  <td className="px-4 border border-gray-200">
+                    {row.capacity ? row.capacity : "-"} / 40
+                  </td>
```
Comment (oohgan) [inline src/pages/PrismaOnline/partials/CourseList.tsx:83]: ini biso di map, kalo datanyo perlu di ubah kayak ditambahin /40, itu logic di custom function buat parsenyo,

atau biarin kayak ini, tapi jangan ternary, pake `row.class || "-"`, trus classname dibikin custom const `commonTdClass`

File: src/pages/ProofOfFilling/partials/SelectedCourseList.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    "waktu kuliah",
+    "kelas",
+    "jumlah sks",
+  ];
+
+  return (
+    <div className="flex flex-col gap-5">
+      <BaseTable>
+        <TableBody header={parentHeader}>
+          {Array.from({ length: 6 }, (_, index) => index + 1).length !== 0 ? (
```
Comment (oohgan) [inline src/pages/ProofOfFilling/partials/SelectedCourseList.tsx:28]: untuk bikin dummy mapping? pake [...Array(4)].map((_, index) => { ... } )

File: src/pages/ProofOfFilling/partials/SelectedCourseTable.tsx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+} from "@components/Table";
+import {
+  getCurrentSKS,
+  getSelectedCourses,
+  leaveCourse,
+} from "../../../api/course/service";
+import { dayName, sessionName, statusDetail, statusName } from "@utils/consts";
+import { cn } from "@utils/cn";
+import { useState } from "react";
+import { SelectedCourseModel } from "../../../api/course/model";
```
Comment (oohgan) [inline src/pages/ProofOfFilling/partials/SelectedCourseTable.tsx:15]: bikin aliases import, jelek soalnyo import pake .. ..

**Review by oohgan — COMMENTED** [review]
bikin pr baru

File: src/common/utils/consts.ts (hunk 3 above)
Comment (Rendy752) [reply src/common/utils/consts.ts:32]: `const statusDetailName: { [key: number]: string } = {
  1: statusNameMap.WAITING,
  2: statusNameMap.APPROVED,
  3: statusNameMap.REJECTED,
  4: statusNameMap.NOT_TAKEN,
};`

`const statusDetailColor = {
  [statusNameMap.WAITING]: "bg-yellow-300",
  [statusNameMap.APPROVED]: "bg-blue-300",
  [statusNameMap.REJECTED]: "bg-red-300",
  [statusNameMap.NOT_TAKEN]: "bg-gray-300",
};`

File: src/pages/PrismaOnline/partials/CourseList.tsx (hunk 4 above)
Comment (Rendy752) [reply src/pages/PrismaOnline/partials/CourseList.tsx:53]: `const { courses, handleJoinCourse, handleLeaveCourse } = useCourses(keyword);`

File: src/pages/ProofOfFilling/partials/SelectedCourseList.tsx (hunk 6 above)
Comment (Rendy752) [reply src/pages/ProofOfFilling/partials/SelectedCourseList.tsx:28]: `{[...Array(6)].length !== 0 ? (
            [...Array(6)].map((_, index) => {}
)}`

File: src/pages/ProofOfFilling/partials/SelectedCourseTable.tsx (hunk 7 above)
Comment (Rendy752) [reply src/pages/ProofOfFilling/partials/SelectedCourseTable.tsx:15]: `"@models/*": ["./src/api/*/model.ts"],`

`import { SelectedCourseModel } from "@models/course";`

File: src/pages/PrismaOnline/partials/CourseList.tsx (hunk 5 above)
Comment (Rendy752) [reply src/pages/PrismaOnline/partials/CourseList.tsx:83]: `const commonTdClass="px-4 border border-gray-200";`

`const tdDataMap: (string | number)[] = [
                index + 1,
                (row.code || "-") + " | " + (row.name || "-"),
                row.class || "-",
                row.sks + " sks" || "-",
                row.semester_id || "-",
                row.capacity + " / 40" || "-",

                row.schedule
                  ? row.schedule
                      .map(
                        (schedule) =>
                          `${dayName[schedule.day_id] || "?"} ${sessionName[schedule.session_id] || "000 - 00:00"}`,
                      )
                      .join(", ")
                  : "-",
                statusName || "-",

                row.capacity >= 40
                  ? "Penuh"
                  : statusName == statusNameMap.APPROVED ||
                      statusName == statusNameMap.WAITING
                    ? "Keluar kelas"
                    : "Masuk kelas",
              ];`

**Pull request merged**